Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1632,38 +1808,40 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
 
-    Vector<GridTrack>& allTracks = tracks(m_direction);
-    Vector<GridItemWithSpan> itemsSortedByIncreasingSpan;
-    Vector<GridItemWithSpan> itemsCrossingFlexibleTracks;
-    SingleThreadWeakHashSet<RenderBox> itemsSet;
-    MasonryIndefiniteItems masonryIndefiniteItems;
+    StdMap<unsigned, MasonryMinMaxTrackSize> indefiniteTrackSizes;
+    StdMultimap<unsigned, GridItemWithSpan> definiteMultiTrackItems;
+    StdMultimap<unsigned, GridItemWithSpan> definiteMultiTrackItemsWithFlex;
 
-    if (m_grid.hasGridItems()) {
-        computeIndefiniteItemsForMasonry(masonryIndefiniteItems, gridLayoutState);
+    // Compute Span Sizes
+    computeIndefiniteItemsForMasonry(indefiniteTrackSizes, definiteMultiTrackItems, definiteMultiTrackItemsWithFlex, gridLayoutState);
```

Comment:
We should consider if we can try to retrun something here by value instead I think that would make it a bit cleaner

Reply:
This would need to return 3 different items 🤔

Reply:
We can create a struct or another form to hold the return, but the current approach feels simpler to me.